php/index.php

```diff
@@ -20,91 +27,123 @@ function __construct($config)
+            ->withStatus($status);
     }
+}
+
 
-    function success($status, $result, $contenttype = 'application/json')
+class JsonBodyParserMiddleware implements MiddlewareInterface
+{
+    // https://www.slimframework.com/docs/v4/objects/request.html#the-request-body
+    public function process(Request $request, RequestHandler $handler): Response
     {
-        $this->response->headers->set('Content-Type', $contenttype);
-        $this->response->setStatus($status);
-        $this->response->write(json_encode($result));
+        $contentType = $request->getHeaderLine('Content-Type');
+
+        if (strstr($contentType, 'application/json')) {
+            $contents = json_decode(file_get_contents('php://input'), true);
+            if (json_last_error() === JSON_ERROR_NONE) {
+                $request = $request->withParsedBody($contents);
+            }
+        }
+        return $handler->handle($request);
     }
 }
 
-$app = new CrateResource($config);
-// apply CORS headers to all responses
-$app->add(new \CorsSlim\CorsSlim());
+
+$container = new Container();
+AppFactory::setContainer($container);
+
+$container->set('helper', function () {
+    $config = parse_ini_file('app.ini');
+    return new AppHelper($config);
+});
+
+$app = AppFactory::create();
+$app->add(JsonBodyParserMiddleware::class);
 
 /**
  * Default action.
  */
-$app->get('/', function() use ($app)
+$app->get('/', function (Request $req, Response $res, $args = [])
 {
-    $app->success(200, 'Server up and running');
-})->name('default');
+    $helper = $this->get('helper');
+    return $helper->success($res, 200, 'Server up and running');
+})->setName('default');
 
 /**
  * Get the post for a given id.
  */
-$app->get('/post/:id', function($id) use ($app)
+$app->get('/post/{id}', function (Request $req, Response $res, $args = [])
 {
-    $qry = $app->conn->prepare("SELECT p.*, c.name as country, c.geometry as area
+    $helper = $this->get('helper');
+    $id = $args["id"];
+    $qry = $helper->conn->prepare("SELECT p.*, c.name as country, c.geometry as area
             FROM guestbook.posts AS p, guestbook.countries AS c
             WHERE within(p.\"user\"['location'], c.geometry) AND p.id = ?");
     $qry->bindParam(1, $id);
     $qry->execute();
     $result = $qry->fetch(PDO::FETCH_ASSOC);
     if (!$result) {
-        $app->not_found("Post with id=\"{$id}\" not found");
+        return $helper->not_found($res, "Post with id=\"{$id}\" not found");
     } else {
-        $app->success(200, $result);
+        return $helper->success($res, 200, $result);
     }
-})->name('post-get');
+})->setName('post-get');
+
 /**
- * insert a posts.
+ * Insert a post.
  */
-$app->post('/posts', function() use ($app)
+$app->post('/posts', function (Request $req, Response $res, $args = [])
 {
-    $data      = json_decode($app->request->getBody());
-    $user      = $data->user;
-    $text      = $data->text;
-    $image_ref = $data->image_ref;
+    $helper = $this->get('helper');
```

Comment:
Indentation here should be the same as for the lines below?

Reply:
Thanks. I've addressed it with 39b07742a.